featuretools/computational_backends/pandas_backend.py

```diff
@@ -210,6 +243,72 @@ def generate_default_df(self, instance_ids, extra_columns=None):
                     default_df[c] = [np.nan] * len(instance_ids)
         return default_df
 
+    def _find_necessary_columns(self, only_if_needs_all_values=False):
```

Comment:
is `only_if_needs_all_values` an optimization or is it required for this to work? a bit hard to follow its role. in general, I find it confusing when a function has a single boolean parameter that controls behavior like this unless it is very obvious what it does. as far as I can tell, it there is only one place in the code where we set it to true. if that is the case, I'd prefer if the optimization happens more explicitly outside the function. makes the function itself more readable and maintainable.